storage/postgresStorage.go

```diff
@@ -0,0 +1,476 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"doccer/model"
+	_ "github.com/lib/pq"
+	"strconv"
+)
+
+type PostgresStorage struct {
+	Dbc *sql.DB
+}
+
+func (p *PostgresStorage) ClearAllTables() {
+	_, _ = p.Dbc.Exec("TRUNCATE Users, DocGroupRestriction, DocMemberRestriction, Docs, GroupMember, GeneralInfo, Groups1, Password CASCADE ;")
+	_, _ = p.Dbc.Exec("insert into GeneralInfo values (0, 0, 0, 0)")
+}
+
+func (p *PostgresStorage) GenerateNewUserId() model.Id {
+	ctx := context.Background()
+	tx, _ := p.Dbc.BeginTx(ctx, nil)
+
+	row := tx.QueryRow("select g.last_user_id from GeneralInfo g where g.base_id = 0")
+
+	lastId := 0
+	_ = row.Scan(&lastId)
+	id := strconv.Itoa(lastId)
+	lastId += 1
+
+	_, _ = tx.ExecContext(ctx, "update GeneralInfo set last_user_id = $1 where base_id = 0", lastId)
+
+	_ = tx.Commit()
```

Comment:
Do you see any concurrency risks here?

Reply:
What happens when two concurrent transactions assign the same user id?

Reply:
Added mutexes to fix concurrency